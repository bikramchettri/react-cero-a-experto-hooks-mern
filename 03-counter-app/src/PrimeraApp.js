import React, { Fragment } from 'react';
import PropTypes from 'prop-types'

//Funcional components
const PrimeraApp = ({saludar}) => {
    const saludo =  'Hola mundo desde una constante';
    const objeto = {
        nombre: 'Juan',
        edad: 28
    }
    
    return (
        <> {/* los fragment se pueden poner omitiendo el texto Fragment*/}
            <h1>{saludo}</h1>
            <pre>objeto {JSON.stringify(objeto, null, 3)}</pre>
            <p>Mi primera aplicaci√≥n</p>
            <p>props {saludar}</p>
        </>
    );
}

PrimeraApp.propType = {
    saludar: PropTypes.string.isRequired
}
export default PrimeraApp;