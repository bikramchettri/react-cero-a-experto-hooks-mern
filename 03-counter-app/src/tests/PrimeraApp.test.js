import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <PrimeraApp/>', ()=> {
    /* test('deben de moestrar el mensaje "hola soy..."', () => {
         const saludo = 'hola soy...';
     
         const { getByText } = render( <PrimeraApp saludo={saludo} /> )
         
         expect( getByText(saludo)).toBeInTheDocument();
     }) */
     test('debe de mostrar  <PrimeraApp/> correctamente ', () => {
        const saludo = 'hola soy...';

        const wrapper = shallow( <PrimeraApp saludo={saludo}/>);

        expect(wrapper).toMatchSnapshot();

     })
     
})