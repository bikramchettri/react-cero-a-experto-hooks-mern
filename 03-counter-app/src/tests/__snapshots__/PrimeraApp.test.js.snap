// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> debe de mostrar  <PrimeraApp/> correctamente  1`] = `
<Fragment>
   
  <h1>
    hola soy...
  </h1>
  <h3>
    soy el subtitulo
  </h3>
</Fragment>
`;
